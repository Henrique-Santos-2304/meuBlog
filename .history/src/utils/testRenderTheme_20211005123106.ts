import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<ThemeProvider>{children}</ThemeProvider> );
};
