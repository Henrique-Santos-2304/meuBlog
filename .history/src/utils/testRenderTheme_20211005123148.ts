import { ThemeProvider } from "styled-components";
import { render, RenderResult } from "@testing-library/react";
import theme from "../styles/theme";
export const renderTheme = (children: React.ReactElement): RenderResult => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider> );
};
