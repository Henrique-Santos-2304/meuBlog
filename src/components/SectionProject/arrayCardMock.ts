export const images = [
  {
    title: "LeScone",
    img: "https://res.cloudinary.com/defnibbpl/image/upload/v1634678383/Lescone_9fb07ad59b.gif",
    buttonCode: "https://github.com/Henrique-Santos-2304/Restaurante-LeScone",
  },
  {
    title: "Multitech",
    img: "https://res.cloudinary.com/defnibbpl/image/upload/v1634678422/Multitech_desktop_90fabbf476_83baa085c2.gif",
    buttonCode: "https://github.com/Henrique-Santos-2304/Multitech",
  },
  {
    title: "HdcAgency",
    img: "https://res.cloudinary.com/defnibbpl/image/upload/v1634845622/hdc_Agency_f400e88579.gif",
    buttonCode:
      "https://github.com/Henrique-Santos-2304/Agencia-de-Desenvolvedores",
  },
];
