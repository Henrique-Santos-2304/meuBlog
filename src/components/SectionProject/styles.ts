import styled from "styled-components";

export const Wrapper = styled.div`
  margin-top: 22rem;
  height: 84vh;
`;
