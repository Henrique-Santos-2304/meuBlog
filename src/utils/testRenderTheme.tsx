import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
