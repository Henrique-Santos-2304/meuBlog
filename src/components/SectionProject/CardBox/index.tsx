import * as S from "./styles";
import Link from "next/link";

type cardsSlides = {
  title: string;
  img: string;
  buttonCode: string;
  marginBox: string;
};
const CardBox = ({ title, img, buttonCode, marginBox }: cardsSlides) => (
  <S.ContentSection marginBox={marginBox} aria-label="Titulo do projeto">
    <S.ContentTitle>
      <S.TitleProject>{title}</S.TitleProject>
    </S.ContentTitle>
    <S.Wrapper aria-label="Imagem do projeto">
      <S.PhotoProject src={img} layout="fill" quality={100} />
    </S.Wrapper>
    <S.ContentButton>
      <Link href={buttonCode} passHref>
        <a target="_blank">
          <S.ButtonCode aria-label="Link para código do Projeto">
            <S.Icon />
            Ver Código
          </S.ButtonCode>
        </a>
      </Link>
    </S.ContentButton>
  </S.ContentSection>
);

export default CardBox;
