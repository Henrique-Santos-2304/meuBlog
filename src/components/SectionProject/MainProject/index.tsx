/* eslint-disable @typescript-eslint/no-explicit-any */
import * as S from "./styles";

import Headings from "components/globalComponents/Headings";
import { images } from "../arrayCardMock";
import SliderPhoto from "components/globalComponents/SliderPhoto";

const SectionProject = () => {
  return (
    <S.Wrapper id="projects">
      <S.Title>
        <Headings
          msg="Meus Projetos"
          fontSizeText="2rem"
          marginText="1.5rem 0 1.4rem 3rem"
          aria-label="Titulo da seção"
        />
      </S.Title>
      <S.ContentSlide>
        <SliderPhoto
          images={images}
          receptorCards="projects"
          slidesToShow={1}
          marginBox="2rem 0 1rem"
        />
      </S.ContentSlide>
    </S.Wrapper>
  );
};

export default SectionProject;
