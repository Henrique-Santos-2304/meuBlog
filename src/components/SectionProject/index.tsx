import * as S from "./styles";

const SectionProject = () => (
  <S.Wrapper id="project">
    <h1>SectionProject</h1>
  </S.Wrapper>
);

export default SectionProject;
