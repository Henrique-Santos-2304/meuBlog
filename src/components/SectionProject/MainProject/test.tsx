import "match-media-mock";
import SectionProject from ".";

import { renderTheme } from "utils/testRenderTheme";

describe("<SectionProject />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<SectionProject />);
    expect(container).toMatchSnapshot();
  });
});
