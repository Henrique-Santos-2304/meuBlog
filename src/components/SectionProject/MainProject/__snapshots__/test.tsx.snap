// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionProject /> should render  1`] = `
.c0 {
  max-width: 74rem;
  margin: 0 auto;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  padding: 1rem 0 3rem;
}

.c3 {
  width: 100%;
  margin: 1rem auto;
}

.c2 {
  position: relative;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  color: #f8f8f8;
  padding: 0;
  margin: 1.5rem 0 1.4rem 3rem;
  font-size: 2rem;
  font-family: "Roboto",monospace;
}

.c2::before {
  content: "";
  position: absolute;
  top: 50%;
  left: -1.5rem;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 0.4rem;
  height: 90%;
  border-radius: 2rem;
  background: #b7f600;
}

.c5 {
  max-width: (100%,74rem);
  margin: 2rem 0 1rem;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  width: 100%;
  padding: 0.9rem 0;
  background: rgba(0,0,0,0.84);
}

.c7 {
  text-align: center;
  font-size: 1.8rem;
  color: #b7f600;
  font-style: italic;
}

.c8 {
  width: 100%;
  height: 20rem;
  position: relative;
}

.c9 {
  width: 100%;
  padding: 1rem 0;
  background: rgba(0,0,0,0.84);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.4rem;
  font-weight: 700;
  background: transparent;
  border: 0.1rem solid rgba(255,255,255,0.15);
  color: #f8f8f8;
  padding: 0.6rem 1.5rem;
  border-radius: 1rem;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
  cursor: pointer;
}

.c10:hover {
  -webkit-filter: brightness(0.7);
  filter: brightness(0.7);
  border: 0.1rem solid #f8f8f8;
}

.c11 {
  margin-right: 0.8rem;
  font-size: 1.8rem;
}

.c4 {
  position: relative;
}

.c12 {
  bottom: -3rem;
  cursor: pointer;
}

.c12 ul:last-child {
  color: #b7f600;
}

.c12 ul li {
  color: #b7f600;
}

.c12 ul li.slick-active button::before {
  color: #b7f600;
}

.c12 ul li button::before {
  font-size: 2.3rem;
}

@media (min-width:620px) {
  .c3 {
    padding: 0 2.4rem;
    border-radius: 2rem;
  }
}

@media (min-width:768px) {
  .c3 {
    padding: 0 4.0rem;
  }
}

@media (min-width:620px) {
  .c1 {
    padding: 0 2.4rem;
    margin: 0 auto;
  }
}

@media (min-width:768px) {
  .c1 {
    padding: 0 4.0rem;
  }
}

@media (min-width:620px) {
  .c5 {
    border-radius: 2rem;
  }
}

@media (min-width:768px) {
  .c5 {
    padding: 0 8rem;
  }
}

@media (min-width:620px) {
  .c6 {
    border-radius: 1rem 1rem 0 0;
  }
}

@media (min-width:620px) {
  .c9 {
    border-radius: 0 0 1rem 1rem;
  }
}

@media (min-width:620px) {
  .c4 {
    border-radius: 2rem;
  }
}

<div>
  <section
    class="c0"
    id="projects"
  >
    <header
      class="c1"
    >
      <h1
        class="c2"
      >
        meus projetos
      </h1>
    </header>
    <section
      class="c3"
    >
      <div
        class="slick-slider c4 slick-initialized"
        dir="ltr"
      >
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          >
            <div
              aria-hidden="false"
              class="slick-slide slick-active slick-current"
              data-index="0"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  aria-label="Linguagens,Frameworks ou bibliotecas dominadas"
                  class="c5"
                >
                  <header
                    class="c6"
                  >
                    <h1
                      class="c7"
                    >
                      Titulo da imagem
                    </h1>
                  </header>
                  <div
                    aria-label="Imagem do projeto"
                    class="c8"
                  >
                    <div
                      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                    >
                      <img
                        alt="logo da foto1"
                        class=""
                        data-nimg="fill"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                      />
                      <noscript />
                    </div>
                  </div>
                  <footer
                    class="c9"
                  >
                    <a
                      href="http://projeto1.com"
                      target="_blank"
                    >
                      <button
                        aria-label="link para codigo"
                        class="c10"
                      >
                        <svg
                          class="c11"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 496 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                          />
                        </svg>
                        ver codigo
                      </button>
                    </a>
                  </footer>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              class="slick-slide"
              data-index="1"
              style="outline: none; width: 0px;"
              tabindex="-1"
            >
              <div>
                <div
                  aria-label="Linguagens,Frameworks ou bibliotecas dominadas"
                  class="c5"
                >
                  <header
                    class="c6"
                  >
                    <h1
                      class="c7"
                    >
                      titulo da foto
                    </h1>
                  </header>
                  <div
                    aria-label="Imagem do projeto"
                    class="c8"
                  >
                    <div
                      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
                    >
                      <img
                        alt="logo da foto2"
                        class=""
                        data-nimg="fill"
                        decoding="async"
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                      />
                      <noscript />
                    </div>
                  </div>
                  <footer
                    class="c9"
                  >
                    <a
                      href="http://projeto2.com"
                      target="_blank"
                    >
                      <button
                        aria-label="link para codigo"
                        class="c10"
                      >
                        <svg
                          class="c11"
                          fill="currentColor"
                          height="1em"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 496 512"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                          />
                        </svg>
                        ver codigo
                      </button>
                    </a>
                  </footer>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c12 slick-dots"
        >
          <ul
            aria-label="Botões para trocar as fotos"
          >
             
            <li
              class="slick-active"
            >
              <button>
                1
              </button>
            </li>
            <li
              class=""
            >
              <button>
                2
              </button>
            </li>
             
          </ul>
        </div>
      </div>
    </section>
  </section>
</div>
`;
